SonarQube.Common/AnalysisProperties/FilePropertyProvider.cs

```diff
@@ -75,6 +75,22 @@ public static bool TryCreateProvider(IEnumerable<ArgumentInstance> commandLineAr
             return false;
         }
 
+        public static FilePropertyProvider Load(string filePath)
+        {
+            if (string.IsNullOrWhiteSpace(filePath))
+            {
+                throw new ArgumentNullException("filePath");
+            }
+            if (!File.Exists(filePath))
+            {
+                throw new FileNotFoundException(Resources.ERROR_SettingsFileNotFound, filePath);
+            }
+
+            AnalysisProperties properties = AnalysisProperties.Load(filePath);
+            FilePropertyProvider provider = new FilePropertyProvider(properties, false);
+            return provider;
+        }
+
```

Comment:
The FilePropertyProvider is over-complicated and possibly redundant. We should look at separating out the file locating logic and see if what is left is worth keeping. Not part of this review, though.